databases/postgresql16-dev/Portfile

```diff
@@ -0,0 +1,180 @@
+                    --libdir=${libdir} \
+                    --includedir=${prefix}/include/${name} \
+                    --datadir=${prefix}/share/${name} \
+                    --docdir=${prefix}/share/doc/${name} \
+                    --mandir=${prefix}/share/man \
+                    --with-includes=${prefix}/include \
+                    --with-libraries=${prefix}/lib \
+                    --with-openssl \
+                    --with-bonjour \
+                    --with-readline \
+                    --with-zlib \
+                    --with-lz4 \
+                    --with-zstd \
+                    --with-libxml \
+                    --with-libxslt \
+                    --with-gssapi \
+                    --enable-thread-safety \
+                    --with-uuid=e2fs \
+                    --with-icu
+
+configure.ldflags-append    -headerpad_max_install_names
+configure.universal_args-delete --disable-dependency-tracking
+
+# building psql with clang from Xcode prior to 4.4 causes segfault on query; see #31717
+compiler.blacklist-append {clang < 421}
+compilers.choose          cc cxx
+compilers.setup           -gcc -fortran -clang15 -clang33 -clang34 -clang37
+notes "To use the postgresql server as a startup item install the ${name}-server port"
+
+if {![info exists universal_possible]} {
+    set universal_possible [expr {${os.universal_supported} && [llength ${configure.universal_archs}] >= 2}]
+}
+if {${universal_possible} && [variant_isset universal]} {
+    configure.cflags-append    -DAC_APPLE_UNIVERSAL_BUILD
+}
+
+build.type          gnu
+build.target        world
+
+destroot.target     install-world
+
+test.run            yes
+test.target         check
+
+# remember to change livecheck with first proper release
+livecheck.type      regex
+#livecheck.regex    (16\\.\[.0-9\]*\[a-z\]+\[0-9\])
+livecheck.regex    (16\[a-z\]+\[0-9\])
+#livecheck.regex    (16\\.\[.0-9\]+)
+livecheck.url       ${homepage}/ftp/source/
+
+post-destroot {
+    system "cd ${destroot}${prefix}/bin && ln -sf ${libdir}/bin/psql psql16"
+
+    reinplace -E "s#^CFLAGS =#CFLAGS +=#g" \
+        ${destroot}${prefix}/lib/${name}/pgxs/src/Makefile.global
+}
+
+if {!${universal_possible} || ![variant_isset universal]} {
+    post-destroot {
+        reinplace -E "s#-arch\[\[:space:\]\]+\[\[:alnum:\]_\]+##g" \
+            ${destroot}${prefix}/lib/${name}/pgxs/src/Makefile.global
+    }
+} else {
+    merger-post-destroot {
+        foreach arch ${configure.universal_archs} {
+            reinplace -E "s#-arch\[\[:space:\]\]+\[\[:alnum:\]_\]+##g" \
+                ${destroot}-${arch}${prefix}/lib/${name}/pgxs/src/Makefile.global
+
+            reinplace -E "s#${worksrcpath}-${arch}#${worksrcpath}#g" \
+                ${destroot}-${arch}${prefix}/lib/${name}/pgxs/src/Makefile.global
+        }
+    }
+}
+
+select.group        postgresql
+select.file         ${filespath}/${name}
+
+variant python requires python3 description {add support for python} {
+}
```

Comment:
It doesn't seem like we need two variants that do the same thing. Presumably the variant used to be called "python" and was then changed to "python3" and the python variant was retained to assist users upgrading from the previous variant name. No such assistance is needed here since this is a new port.